Menu & User Settings

GuilhermeHideki opened the pull request:
Created a menu and moved some options to load outside the code.
Needs probably an user setting

GuilhermeHideki committed:
+ User Settings
+ Menu

GuilhermeHideki committed:
snake_case

adrian17 commented:
...ah, now I remember why I didn't try saving the config in a .sublime-settings file - because you can't have multiline strings in JSON and that makes writing a bigger template there almost impossible. Any ideas?

GuilhermeHideki commented:
I would use an `insert snippet` command, using a `sublime-snippet` file. I didn't dig into the docs, but it could work, I think.

You could also create an array, and parse each line, like

``` json5
"filename_content":
    [
        "{comment_line} {challenge_title}",
        "{comment_line} {challenge_url}",
        "",
        "def main():",
        "{t}pass", // {t} is \t
        "",
        "",
        "if __name__ == \"__main__\":",
        "{t}main()"
    ],
```

and 3 "replace dictionaries", for filename, readme and both

GuilhermeHideki committed:
Add default config file


User needs only need to set a `{ "path": "mypath" }` if he wants default options
